pkg/controllers/networkpolicy/pod_handler.go

```diff
@@ -0,0 +1,186 @@
+)
+
+// EnqueueRequestForPod handles Pod events and triggers NetworkPolicy reconciliation
+// when Pods with named ports are created, updated, or deleted.
+type EnqueueRequestForPod struct {
+	Client                  client.Client
+	NetworkPolicyReconciler *NetworkPolicyReconciler
+}
+
+func (e *EnqueueRequestForPod) Create(_ context.Context, createEvent event.CreateEvent, q workqueue.TypedRateLimitingInterface[reconcile.Request]) {
+	e.Raw(createEvent, q)
+}
+
+func (e *EnqueueRequestForPod) Update(_ context.Context, updateEvent event.UpdateEvent, q workqueue.TypedRateLimitingInterface[reconcile.Request]) {
+	e.Raw(updateEvent, q)
+}
+
+func (e *EnqueueRequestForPod) Delete(_ context.Context, deleteEvent event.DeleteEvent, q workqueue.TypedRateLimitingInterface[reconcile.Request]) {
+	e.Raw(deleteEvent, q)
+}
+
+func (e *EnqueueRequestForPod) Generic(_ context.Context, genericEvent event.GenericEvent, q workqueue.TypedRateLimitingInterface[reconcile.Request]) {
+	e.Raw(genericEvent, q)
+}
+
+func (e *EnqueueRequestForPod) Raw(evt interface{}, q workqueue.TypedRateLimitingInterface[reconcile.Request]) {
+	var pods []v1.Pod
+	var obj client.Object
+
+	switch et := evt.(type) {
+	case event.CreateEvent:
+		obj = et.Object.(*v1.Pod)
+	case event.UpdateEvent:
+		obj = et.ObjectNew.(*v1.Pod)
+	case event.DeleteEvent:
+		obj = et.Object.(*v1.Pod)
+	case event.GenericEvent:
+		obj = et.Object.(*v1.Pod)
+	default:
+		log.Error(nil, "Unknown event type", "event", evt)
+		return
+	}
+
+	pod := obj.(*v1.Pod)
+	if isInSysNs, err := util.IsSystemNamespace(e.Client, pod.Namespace, nil); err != nil {
+		log.Error(err, "Failed to fetch namespace", "namespace", pod.Namespace)
+		return
+	} else if isInSysNs {
+		log.V(2).Info("POD is in system namespace, do nothing")
+		return
+	}
+	pods = append(pods, *pod)
+	err := reconcileNetworkPolicy(e.Client, pods, q)
+	if err != nil {
+		log.Error(err, "Failed to reconcile network policy")
+	}
+}
+
+func getAllPodPortNames(pods []v1.Pod) sets.Set[string] {
+	podPortNames := sets.New[string]()
+	for _, pod := range pods {
+		for _, container := range pod.Spec.Containers {
+			for _, port := range container.Ports {
+				if port.Name != "" {
+					podPortNames.Insert(port.Name)
+				}
+			}
+		}
+	}
+	return podPortNames
+}
+
+// reconcileNetworkPolicy is triggered by Pod events to reconcile NetworkPolicies with named ports
+func reconcileNetworkPolicy(pkgClient client.Client, pods []v1.Pod, q workqueue.TypedRateLimitingInterface[reconcile.Request]) error {
+	podPortNames := getAllPodPortNames(pods)
+	log.V(1).Info("POD named port", "podPortNames", podPortNames)
+
+	npList := &networkingv1.NetworkPolicyList{}
+	err := pkgClient.List(context.Background(), npList)
+	if err != nil {
+		log.Error(err, "Failed to list all the network policies")
+		return err
+	}
+
+	for i := range npList.Items {
+		np := &npList.Items[i]
+		shouldReconcile := false
+
+		// Check ingress rules for named ports
```

Comment:
we can wrap this reconcile checking condition in a func, so that namespace handler can reuse this func to check for namedport.